exp/controllers/awsmachinepool_controller.go

```diff
@@ -255,7 +258,22 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(_ context.Context, machinePoo
 	}
 
 	// Make sure Spec.ProviderID is always set.
-	// machinePoolScope.SetProviderID(instance.ID, instance.AvailabilityZone)
+	machinePoolScope.AWSMachinePool.Spec.ProviderID = fmt.Sprintf("%s", asg.ID)
+	providerIDList := make([]string, len(asg.Instances))
+
+	for i, ec2 := range asg.Instances {
+		providerIDList[i] = fmt.Sprintf("aws:///%s/%s", ec2.AvailabilityZone, ec2.ID)
+		// if ec2.State == infrav1.InstanceStateRunning {
+		// 	readyCount++
+		// }
```

Comment:
Was this intentionally commented out?

Reply:
yeah. `ec2.State` isn't being set anywhere. i'll cut another ticket to brush `AWSMachinePool.Status` as a whole later which will include setting status of each instance